spinn_utilities/config_holder.py

```diff
@@ -0,0 +1,285 @@
+
+def config_options(section):
+    """Return a list of option names for the given section name.
+
+    :param str section: What section to list options for.
+    """
+    return __config.options(section)
+
+
+def _check_lines(py_path, line, lines, index, method):
+    """
+    Support for check_python_file. Gets section and option name
+
+    :param str line: Line with get_config call
+    :param list(str) lines: All lines in the file
+    :param int index: index of line with get_config call
+    :raise Exception: If an unexpected or uncovered get_config found
+    """
+    while ")" not in line:
+        index += 1
+        line += lines[index]
+    parts = line[line.find("(", line.find("get_config")) + 1:
+                 line.find(")")].split(",")
+    section = parts[0].strip().replace("'", "").replace('"', '')
+    option = parts[1].strip()
+    if option[0] == "'":
+        option = option.replace("'", "")
+    elif option[0] == '"':
+        option = option.replace('"', '')
+    else:
+        print(line)
+        return
+    try:
+        method(section, option)
+    except Exception:
+        raise Exception(f"failed in line:{index} of file: {py_path} with "
+                        f"section:{section} option:{option}")
+
+
+def check_python_file(py_path):
+    """
+    A testing function to check that all the get_config calls work
+
+    :param str py_path: path to file to be checked
+    :raise Exception: If an unexpected or uncovered get_config found
+    """
+    with open(py_path, 'r') as py_file:
+        lines = py_file.readlines()
+        for index, line in enumerate(lines):
+            if "get_config_bool(" in line:
+                _check_lines(py_path, line, lines, index, get_config_bool)
+            if "get_config_float(" in line:
+                _check_lines(py_path, line, lines, index, get_config_float)
+            if "get_config_int(" in line:
+                _check_lines(py_path, line, lines, index, get_config_int)
+            if "get_config_str(" in line:
+                _check_lines(py_path, line, lines, index, get_config_str)
+            if "get_config_str_list(" in line:
+                _check_lines(py_path, line, lines, index, get_config_str_list)
+
+
+def find_double_defaults():
+    """
+    A testing function to find options in multiple default cfg files
+
+    :raise Exception: If an option is repeated.
+    """
+    parser = CamelCaseConfigParser()
+    options = set()
+    for default in __default_config_files:
+        with open(default) as reader:
+            for line in reader.readlines():
+                stripped = line.strip()
+                if len(stripped) == 0:
+                    continue
+                if stripped.startswith("#"):
+                    continue
+                if stripped.startswith("["):
+                    continue
+                if "Overwritten in Specific config file" in stripped:
+                    continue
+                option = stripped.split(" ")[0]
```

Comment:
It is possible to read config files and iterate their contents.  Would this be better than manual parsing?

Reply:
If you read the config in it silently overwrites what is found previously.

Which is great if you want the user cfg to overwrite the default and then the cfg in the script directory to overwrite that,

In this case I want to catch any cases where different layers of defaults have the same option.
Except in the specific cases where we expect overides

Reply:
That is true if you use the same ConfigParser, but you can create a new ConfigParser and call read on each run and it shouldn't overwrite anything.

However I now see that you are also looking for a specific comment, so yes, I guess this has to be a custom reader.